### ISSUE TITLE
[Train] add logging to `finish_training` for existing `Callback`s

### ISSUE THREAD

**Issue by matthewdeng** [ray-project/ray#22754]
**Goal:** Add helpful logging to `finish_training` so that the user can easily access results from their Callbacks.

**Example:** The `TBXLoggerCallback` will write to a TensorBoard file. `finish_training` could include a message with instructions on how to open the result  (e.g. "Run `tensorboard --logdir <generated_path>`").

**Comment by matthewdeng** [comment]
@JiahaoYao

**Comment by JiahaoYao** [comment]
Adding one printing sentence, and wonder whether I can use print or should I use `logger` instead?

**Comment by matthewdeng** [comment]
I would recommend using `logger` since it's more structured and the user can disable it if desired.

**Comment by JiahaoYao** [comment]
by the way, will other loggers like `MLflowLoggerCallback` also need the same thing?

**Comment by matthewdeng** [comment]
Yep! Of the existing Callbacks, I can see this bringing immediate user value if done for `MLflowLoggerCallback `, `JsonLoggerCallback`, and `TorchTensorboardProfilerCallback`.

**Comment by JiahaoYao** [comment]
One question here is that it seems like it would be good to use ([link](https://github.com/mlflow/mlflow/issues/3317))
```python
logger.info(
    "To view the training results in your browser, run `mlflow ui --backend-store-uri file:///%s --port 5000`" % self.logdir
)
```
for the `mlflow`. How can I verify this? any minimal example to debug?

Also added, the `finish_training` for `JsonLoggerCallback`.

**Comment by JiahaoYao** [comment]
I have updated my changes in this issue.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
